core/src/main/java/com/scalar/db/api/DistributedTransactionFactory.java

```diff
@@ -0,0 +1,45 @@
+package com.scalar.db.api;
+
+import com.scalar.db.config.DatabaseConfig;
+
+/**
+ * A factory class that creates {@link DistributedTransactionManager}, {@link
+ * DistributedTransactionAdmin}, and {@link TwoPhaseCommitTransactionManager} instances. Each
+ * transaction manager should implement this class to instantiate its implementations of @{@link
+ * DistributedTransactionManager}, {@link DistributedTransactionAdmin}, and {@link
+ * TwoPhaseCommitTransactionManager}. The implementations are assumed to be loaded by {@link
+ * java.util.ServiceLoader}.
+ */
+public interface DistributedTransactionFactory {
```

Comment:
Introduced `DistributedTransactionFactory` and added implementations of each transaction manager.